mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h

```diff
@@ -3821,16 +3821,13 @@ struct Conv2DRewritePattern : public OpRewritePattern<T> {
 
       // if the config do not do padding kernel,
       // gemmAPad = weiGemmGGemmKGemmM
-      auto gemmAPad = padFilter(gemmMExtra, gemmNExtra, gemmKExtra, weiGemmGGemmKGemmM, b, 
-		     loc, filterOobCheckDims, nameToDims, filterShape, filterElementType,
-		     g, k, c, yDotSlice, xDotSlice);
+      Value gemmAPad =
```

Comment:
Out of curiosity, why's this a `Value` and not a `TransformOp`?

Reply:
an `miopen::TransformOp` is also a `Value`, I use it just for brevity, and without going completely oblivious with `auto`

Reply:
Ah. I thought there was some fundamental reason we were doing this.

I think that, since we're relying on the fact that the `Value` being passed into the padding functions being a `TransformOp` (or are we? Does it just need to have certain attributes?), it should be named as such in the function, though having it be a `Value` here is fine.

Reply:
@krzysz00 I should have added you as a reviewer to #279 :)
could you suggest a good name to properly name these functions?

Reply:
The function names aren't what I'm complaining about.

What I'd like to know is: when we pass a `Value` into the `padBlah` functions, what conditions does that `Value` have to meet? Does it have to be a `TransformOp`? Can it be any op, so long as certain attributes are set? Are these preconditions clear?

Reply:
These `padBlah` are utility functions used deep inside the implementation with no external users. And to your question, no, the `Value` reference passed into these functions doesn't need to be a `TransformOp` at this moment. If I understand @asleepzzz  's logic correctly, he is trying to "amend" additional `miopen::TransformOp` in case he finds it necessary, so the input `Value` doesn't necessarily have to be an instance of `miopen::TransformOp`.

Reply:
In which case, having the types involved by `Value` makes things clear and won't be needlessly confusing - objection withdrawn